### PR TITLE
Add release coordination schedule and release timeline

### PR DESCRIPTION
**Pull request by laurenfrederick**
Added the schedule of who is doing the release to the documentation (modeled after CesiumJS) and added a brief timeline of release activities.

## Reviewer checklist

Thank you for taking the time to review this PR. By approving a PR you are taking as much responsibility for these changes as the author.

As you review, please go through the checklist below:

- [ ] Review and run all parts of the test plan on this branch and verify it matches expectations.
    - [ ] If the issue is a bug please make sure you can reproduce the bug in the main branch and then checkout this branch to make sure it actually solved the issue.
- [ ] Review the code and make sure you do not have any remaining questions or concerns. You should understand the code change and the chosen approach. If you are not confident or have doubts about the code, please do not hesitate to ask questions.
    - [ ] Code should follow the [C++ Style Guide](https://cesium.com/learn/cesium-native/ref-doc/style-guide.html)
- [ ] Review the unit tests and make sure there are no missing tests or edge cases.
- [ ] Review documentation changes and updates to `CHANGES.md` to make sure they accurately cover the work in this PR.
- [ ] Verify that the [Contributor License Agreement](https://github.com/CesiumGS/community/tree/main/CLAs) has been submitted, if needed.

### CODE REVIEW

File: doc/topics/release-process.md [hunk 1]
```diff
@@ -30,3 +30,25 @@ A CI job automatically publishes the documentation to the web site at https://ce
...
+## Release Coordinator
+
+| Date      | User             |
+| --------- | ---------------- |
+| 6/1/2026  | `@timoore`       |
+| 7/1/2026  | `@azrogers`      |
+| 8/3/2026  | `@david-lively`  |
+| 9/1/2026  | `@j9liu`         |
+| 10/1/2026 | `@timoore`       |
+| 11/1/2026 | `@azrogers`      |
```
Comment (j9liu) [inline doc/topics/release-process.md]: I believe 11/1 this year is a Sunday!

```suggestion
| 11/2/2026 | `@azrogers`      |
```

File: doc/topics/release-process.md [hunk 2]
```diff
@@ -30,3 +30,25 @@ A CI job automatically publishes the documentation to the web site at https://ce
...
+| 12/1/2026 | `@david-lively`  |
+
+## Cesium Native Runtimes Release Timeline
+
+| Days Until Release | Release Tasks |
+| --- | ---------------- |
+| T-2 | * All pending PRs merged <br> * Begin `cesium-native` release <br> * Update `cesium-native` in all runtimes and perform regression testing |
```
Comment (j9liu) [inline doc/topics/release-process.md]: **NIT:**
```suggestion
| T-2 | * All pending PRs merged in all repos <br> * Begin `cesium-native` release <br> * Update `cesium-native` in all runtimes and perform regression testing |
```

File: doc/topics/release-process.md [hunk 3]
```diff
@@ -30,3 +30,25 @@ A CI job automatically publishes the documentation to the web site at https://ce
...
+| T-1 | * Create commits for `cesium-unity` and `cesium-unreal` <br> * Debug any CI issues |
```
Comment (j9liu) [inline doc/topics/release-process.md]: Small addendum:
```suggestion
| T-1 | * Create release commits for `cesium-unity` and `cesium-unreal` <br> * Debug any CI issues * Prepare GitHub release drafts | 
```

**Review by j9liu — COMMENTED** [review]
Thanks @laurenfrederick! This looks great for the most part, just a few small suggestions!

**Comment by laurenfrederick** [comment]
Thank you, @j9liu! Committed your suggestions.

**Review by j9liu — APPROVED** [review]
Thanks @laurenfrederick!
